src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -993,74 +980,27 @@ public void removeUserFromGroup(User user, String group) {
         userService.removeUserFromGroup(user, group);
     }
 
-    /**
-     * checks if the given group exists in the authentication provider, only on production systems
-     *
-     * @param group the group that should be available
-     */
-    public void checkIfGroupsExists(String group) {
-        if (!Arrays.asList(env.getActiveProfiles()).contains(SPRING_PROFILE_PRODUCTION)) {
-            return;
-        }
-        // only execute this check in the production environment because normal developers (while testing) might not have the right to call this method on the authentication server
-        if (!artemisAuthenticationProvider.isGroupAvailable(group)) {
-            throw new ArtemisAuthenticationException("Cannot save! The group " + group + " does not exist. Please double check the group name!");
-        }
-    }
-
     /**
      * If the corresponding group (student, tutor, editor, instructor) is not defined, this method will create the default group.
      * If the group is defined, it will check that the group exists
      *
      * @param course the course (typically created on the client and not yet existing) for which the groups should be validated
      */
     public void createOrValidateGroups(Course course) {
```

Comment:
Since we do not validate anything anymore I would suggest changing the name to this (or something similar)
```suggestion
     *
     * @param course the course (typically created on the client and not yet existing) for which the groups should be validated
     */
    public void createDefaultGroupsIfNotExisting(Course course) {
```